tests/FirebaseProjectManagerTest.php

```diff
@@ -66,15 +66,43 @@ public function calls_are_passed_to_default_project(): void
     /**
      * @test
      */
-    public function credentials_can_be_configured(): void
+    public function credentials_can_be_configured_using_a_json_file(): void
     {
         // Reference credentials
         $credentialsPath = \realpath(__DIR__ . '/_fixtures/service_account.json');
         $credentials = \json_decode(\file_get_contents($credentialsPath), true);
 
         // Set configuration and retrieve project
         $projectName = 'app';
-        $this->app->config->set('firebase.projects.' . $projectName . '.credentials.file', \realpath(__DIR__ . '/_fixtures/service_account.json'));
+        $this->app->config->set('firebase.projects.' . $projectName . '.credentials', \realpath(__DIR__ . '/_fixtures/service_account.json'));
+        $factory = $this->factoryForProject($projectName);
+
+        // Retrieve service account
+        $serviceAccount = $this->getAccessibleProperty($factory, 'serviceAccount')->getValue($factory);
+
+        // Validate value
+        $this->assertSame($credentials, $serviceAccount);
+    }
```

Comment:
You remove the `.credentials.file` check, I think this test should remain unchanged (except the name) and the new JSON Parsing a new test.